subprojects/core/src/main/java/org/gradle/internal/progress/DefaultBuildOperationExecutor.java

```diff
@@ -151,21 +205,62 @@ public void run(BuildOperationDetails operationDetails, Action<? super BuildOper
         }
     }
 
+    private BuildOperationDescriptor.Builder createDescriptorBuilder(BuildOperation buildOperation, BuildOperationState parent) {
+        long id = nextId.getAndIncrement();
+        BuildOperationDescriptor.Builder descriptorBuilder = buildOperation.description();
+        if (descriptorBuilder == null) {
```

Comment:
Currently we do not have that case (only in tests). Yeah... we probably do not want to allow that.

I remove it and adjust the tests.